### PR TITLE
Ptr-based FBE import ptr-based FBE package

### PR DESCRIPTION
**Pull request by lovio**
This PR makes imported package can be ptr-based FBE.

Now ptr-based FBE can import both ptr-based and template-based FBE.

If you want to import a ptr-based FBE, you should add `--import-ptr` to your fbec scripts

### CODE REVIEW

File: source/generator_cpp.inl [hunk 1]
```diff
@@ -3664,13 +3664,10 @@ std::string GeneratorCpp::ConvertPtrTypeNameAsArgument(const std::string& packag
...
     std::string field_model_type;
-    // Struct
-    if (IsVariantType(p, *field->type)) {
-        // TODO(liuqi): check ptr
-        field_model_type = "FieldModelVariant_" + *p->name + "_" + *field->type;
-    }
-    else if (IsStructType(p, *field->type) && !IsKnownType(*field->type)) {
-        std::string model_name = std::string("FieldModel") + (field->ptr ? "Ptr" : "") + "_" + *p->name + "_" + *field->type;
+    if ((IsStructType(p, *field->type) || IsVariantType(p, *field->type) || ImportPtr()) && !IsKnownType(*field->type)) {
+        // remove . to ::
```
Comment (hurricane1026) [inline source/generator_cpp.inl:3668]: rename？

Comment (lovio) [reply source/generator_cpp.inl:3668]: 对，这里需要把pkg.Expr -> pkg::Expr

Comment (lovio) [reply source/generator_cpp.inl:3668]: 我改一下comment

Comment (lovio) [reply source/generator_cpp.inl]: done

**Review by hurricane1026 — APPROVED** [review]

**Review by hurricane1026 — APPROVED** [review]

**Pull request merged**